src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -247,12 +255,12 @@ private void validateQueryOptions(QueryState queryState, QueryOptions options)
         int nowInSec = options.getNowInSeconds(queryState);
         int userLimit = getLimit(options);
         int userPerPartitionLimit = getPerPartitionLimit(options);
-        int pageSize = options.getPageSize();
+        PageSize pageSize = options.getPageSize();
 
         Selectors selectors = selection.newSelectors(options);
         ReadQuery query = getQuery(queryState, options, selectors.getColumnFilter(), nowInSec, userLimit, userPerPartitionLimit, pageSize);
 
-        if (aggregationSpec == null && (pageSize <= 0 || (query.limits().count() <= pageSize)))
+        if (aggregationSpec == null && !pageSize.isCompleted(query.limits().count(), PageSize.PageUnit.ROWS))
```

Comment:
Should be `PageSize.PageUnit.ROWS` => `pageSize.getUnit()`?

`GuardrailPagingTest.testQueryWithSmallBytePagesWorks` is currently failing, but passes with this change.